[caffe2] Make c10::str works with scoped enum (#152705)

Mizuchi opened the pull request:
Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/152705

Test Plan:
```
buck2 test fbcode//caffe2/c10/test:util_base_tests --fail-fast
```

Differential Revision: D74087796

Mizuchi commented:
@pytorchbot label "topic: not user facing"

zhxchen17 commented:
Looks like the following lint need to be addressed: https://hud.pytorch.org/pr/pytorch/pytorch/152714#41560275911

Otherwise lgtm

Mizuchi committed:
[caffe2] Make c10::str works with scoped enum (#152714)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/152714

Test Plan:
```
buck2 test fbcode//caffe2/c10/test:util_base_tests --fail-fast
```

Reviewed By: praihan

Differential Revision: D74087796
